### DISCUSSION TITLE
Problema con Task 1: Cauchy Matrix

### DISCUSSION THREAD

**Discussion by cristianpenagos** [rramosp/ai4eng.v1#297] [02 - PYTHON]
estoy realizando bien la matriz resultante, pero tengo problemas con el manejo de la excepción, inicialmente lo hice de la siguiente manera:

![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/59f9c7c5-9b97-44f9-9b6d-6fa44631620d)

para lo cual la salida al ejecutar los vectores que contienen la división por 0 para probar el manejo del error arrojan lo siguiente:

![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/782c632d-bd89-4992-96fe-7269edc52799)

seguidamente al intentar enviar el taks a RLXMOOC aparece lo siguiente:
![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/c9f5f413-befe-451a-9a17-28a01ea3d7f8)

intente quitando el If Else que maneja inicialmente que la variable "difference" sea o no 0, provocando que efectivamente se intente realizar la división por 0 y esto dispare una excepción capturada por el try de la siguiente manera:
![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/da35ded1-88dc-414d-94e7-7d787e444c63)
la salida al ejecutar los vectores con división por cero es la siguiente:
![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/b51f7450-a1dd-4c9e-8a9f-df35c93143ad)

y la salida al intentar enviar a la plataforma es esta:
![image](https://github.com/rramosp/ai4eng.v1/assets/24285181/29fd3e47-125b-4160-a111-ad6e0b0aedf9)

¿Cuál es el manejo correcto de la excepción para que la plataforma pueda realizar correctamente las pruebas aleatorias? o qué mal manejo estamos teniendo en este caso?

muchas gracias

**Comment by Felipe-RA** [comment]
Hola @cristianpenagos 

Gracias por compartir tu inquietud por acá, antes de responderte recordarte que en general, en cualquier contexto, es mejor un bloque de código a una captura de pantalla si la herramienta lo soporta. Esto facilita probar tu código!

Ahora, en cuanto a la excepción:

Para esta tarea, sólo se te pide que **alces (`raise`)** la excepción `ValueError`

Veo que en tu función, no sólo la lanzas, sino que también la capturas!

Atrapar y manejar las excepciones usualmente se hace en el `main` de un programa para centralizar este manejo, mientras que las **funcionalidades** se encargan de **identificar** **excepciones** y **lanzarlas**.

En este caso, es el grader quien hace uso de tu función, y trata de identificar que la excepción exista (el grader tiene la estructura `try` `except` y es dentro de ella que lanza tu código)

Espero te aclare tus dudas esto, estoy pendiente!

¡Hasta pronto!
